src/events/cd-event-sessions.vue

```diff
@@ -104,45 +102,54 @@
       deleteChildComponent(index) {
         this.children.splice(index, 1);
       },
-      async loadCurrentUser() {
-        const response = await UserService.getCurrentUser();
-        this.user = response.body.user;
+      async addPhoneNumber() {
+        const userProfile = await UserService.userProfileData(this.user.id);
+        userProfile.phone = this.phone;
+        return UserService.updateUserProfileData(userProfile);
       },
-      //
-      // quantityBooked(val) {
-      //   this.totalBooked = 0;
-      //   Object.keys(tickets).forEach((session) => {
-      //     if (tickets[session].selectedTickets) {
-      //       this.totalBooked += tickets[session].selectedTickets.length;
-      //     }
-      //   });
-      //   this.submitted = false;
-      // },
-      async next() {
-        const validatedChildComponents = await this.validateAllChildComponents();
-        validatedChildComponents.forEach((childIsValid) => {
-          if (!childIsValid) {
-            this.valid = false;
-          } else {
-            this.valid = true;
-          }
+      async addNewChildren() {
+        let promiseChain = Promise.resolve();
+        (this.$refs.allChildComponents).forEach((child) => {
+          promiseChain = promiseChain.then(() => child.createChild());
         });
+        return promiseChain;
+      },
+      bookTickets() {
+        this.$ga.event(this.$route.name, 'click', 'book_tickets', this.totalBooked);
+        return service.manageTickets(this.applications);
+      },
+      async submitForm() {
+        this.errors.clear();
+        if (this.showPhone) {
+          await this.addPhoneNumber();
+        }
+        await this.addNewChildren();
+        this.bookTickets();
+        this.$router.push({ name: 'EventBookingConfirmation',
+          params: { eventId: this.eventId } });
+      },
+      async next() {
+        this.errors.clear();
+        this.valid = false;
+        await this.checkValidatedChildComponents();
 
-        const validPhone = await this.$validator.validateAll();
+        if (this.showPhone) {
+          this.validPhone = await this.$validator.validateAll();
+        }
 
         if (this.valid === false) {
```

Comment:
To avoid the if/else/elseif, define either your default to be the positive result (await this.submitForm) or "any other scenario", which means negating the positive scenario

Reply:
Ok will try that